src/main/java/com/aws/greengrass/disk/spool/DiskSpoolDAO.java

```diff
@@ -51,16 +51,27 @@ public class DiskSpoolDAO {
             RetryUtils.RetryConfig.builder().initialRetryInterval(Duration.ofSeconds(1L))
                     .maxRetryInterval(Duration.ofSeconds(3L)).maxAttempt(3)
                     .retryableExceptions(Collections.singletonList(SQLTransientException.class)).build();
-
+    private static Connection dbConnection;
```

Comment:
should not be static and it should be `volatile`.

We need to think about threads here and ensure that this is all threadsafe.